project/src/main/java/com/google/sps/servlets/EventCreationServlet.java

```diff
@@ -50,8 +60,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String time = request.getParameter("time");
     String description = request.getParameter("description");
     String location = request.getParameter("location");
-    Set<String> labels = Collections.unmodifiableSet(new HashSet<>(
-        Arrays.asList("None"))); // hardcoded for now, we need to create label pool first
+    String text = new StringBuilder().append(name).append(" ").append(description).toString();
+
+    ArrayList<String> categoryNames = new ArrayList<String>();
+
+    // Call NLP API (only works if more than 20 words in text)
+    if (text.trim().split("\\s+").length > 20) {
```

Comment:
Break this chunk out into a new function? That way, instead of a comment, you can have an informative function name

Reply:
done